common/common.cpp

```diff
@@ -418,6 +418,18 @@ bool gpt_params_parse(int argc, char ** argv, gpt_params & params) {
             params.antiprompt.push_back(argv[i]);
         } else if (arg == "--perplexity") {
             params.perplexity = true;
+        } else if (arg == "--ppl-stride") {
+            if (++i >= argc) {
+                invalid_param = true;
+                break;
+            }
+            params.ppl_stride = std::stoi(argv[i]);
+        } else if (arg == "--ppl-output-type") {
+            if (++i >= argc) {
+                invalid_param = true;
+                break;
+            }
+            params.ppl_output_type = std::stoi(argv[i]);
```

Comment:
These two parameters do not appear in the `--help`. I assume this is a simple oversight?

Reply:
I did not add it to the help on purpose. As you can see from the table above, the current implementation is very inefficient. Hence, I decided that for now it is better to have this option kind of hidden from general usage and available only to those who pay attention to the commits. Later, when a better handling of RoPE becomes available (I discussed this with @ggerganov and this is on his radar), the implementation can be improved to be (almost) as efficient as the original `llama.cpp` implementation. My plan was to add the option to the help at that point.

Reply:
Yes, as we discussed - we need more efficient KV cache reuse.
Will track this in the existing issue on the roadmap: https://github.com/ggerganov/llama.cpp/issues/2060

Reply:
At this point, now that the KV changes got merged it should be possible to update the new perplexity calculation to perform about the same as the original. Right?

Reply:
Yes